doc/api/fs.md

```diff
@@ -13,1160 +13,1284 @@ way modeled on standard POSIX functions.
 
 To use this module:
 
-```js
-const fs = require('fs');
+```js esm
+// Using ESM Module syntax:
+import * as fs from 'fs';  // Sync and callback APIs
 ```
 
-All file system operations have synchronous, callback, and promise-based
-forms.
-
-## Synchronous example
-
-The synchronous form blocks the Node.js event loop and further JavaScript
-execution until the operation is complete. Exceptions are thrown immediately
-and can be handled using `try…catch`, or can be allowed to bubble up.
+```js esm
+import * as fs from 'fs/promises';  // Promise APIs
+```
 
-```js
-const fs = require('fs');
+```js cjs
+// Using CommonJS syntax:
+const fs = require('fs');  // Sync and callback APIs
+```
 
-try {
-  fs.unlinkSync('/tmp/hello');
-  console.log('successfully deleted /tmp/hello');
-} catch (err) {
-  // handle the error
-}
+```js cjs
+const fs = require('fs/promises');  // Promise APIs
```

Comment:
The current implementation in https://github.com/nodejs/node/pull/37162 expects to find one ESM block followed by one CJS block (or one CJS block then one ESM one).So you'd need to mode the first CJS block above the second ESM one to get the expected result.

Reply:
Yes, I know. I didn't want to duplicate all of the examples just yet until after that landed.

Reply:
I meant you need to replace:

````markdown
To use this module:

```js esm
// Using ESM Module syntax:
import * as fs from 'fs';  // Sync and callback APIs
```

```js esm
import * as fs from 'fs/promises';  // Promise APIs
```

```js cjs
// Using CommonJS syntax:
const fs = require('fs');  // Sync and callback APIs
```

```js cjs
const fs = require('fs/promises');  // Promise APIs
```
````

With:

````markdown
To use this module:

```js esm
// Using ESM Module syntax:
import * as fs from 'fs';  // Sync and callback APIs
```

```js cjs
// Using CommonJS syntax:
const fs = require('fs');  // Sync and callback APIs
```

```js esm
import * as fs from 'fs/promises';  // Promise APIs
```

```js cjs
const fs = require('fs/promises');  // Promise APIs
```
````